### PR TITLE
removed client calls from workflowIT and generalIT

### PR DESCRIPTION
**Pull request by agduncan94**
https://github.com/dockstore/dockstore/issues/2970

Continuing the process of migrating relevant tests to dockstore_cli.

Note that some tests make sense to have in both repos, so for the ones I kept in this repository I have converted them from using Client.main calls to API calls. The tests are testing the same general ideas, though will not be exactly the same.

I will have a PR in the dockstore_cli repo as well with the new tests.

### CODE REVIEW

**Comment by agduncan94** [comment]
Looks like I am getting that DAG 390 vs 391 issues again.

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryLabelHelper.java  9
+ dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoTagResource.java  27
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/QuayImageRegistry.java  11
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/BitBucketSourceCodeRepo.java  26
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java  2
+ dockstore-common/src/main/java/io/dockstore/common/PipHelper.java  68
+ dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tag.java  11
+ dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ResourceUtilities.java  9
+ dockstore-webservice/src/main/java/io/dockstore/webservice/core/Image.java  39
+ dockstore-webservice/src/main/java/io/dockstore/webservice/core/Checksum.java  46
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoFactory.java  9
         

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoTagResource.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/dockstore/dockstore/pullRequest?prid=4535813&bid=15263544)

**Comment by coverbeck** [comment]
> Looks like I am getting that DAG 390 vs 391 issues again.

Sigh... I spent a little time trying to reproduce it, but couldn't.

1. It usually goes away if you run it again.
2. I wonder if I should change the test to check for greater than 389 and then log a ticket to narrow it down.

File: dockstore-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java [hunk 1]
```diff
@@ -91,6 +67,28 @@
 import org.junit.experimental.categories.Category;
 import org.junit.rules.ExpectedException;
 
+import javax.ws.rs.core.GenericType;
```
Comment (coverbeck) [inline dockstore-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java]: Why did these imports move down? I think this leads to the issue where when we build source files end up getting modified.

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoTagResource.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/dockstore/dockstore/pullRequest?prid=4535813&bid=15263544)

File: dockstore-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java (hunk 1 above)
Comment (agduncan94) [reply dockstore-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java]: The order of my imports in intellij is java, javax, all other imports, all other static imports

**Pull request merged**